src/LibraryManager.Vsix/UI/Models/InstallDialogViewModel.cs

```diff
@@ -372,29 +387,160 @@ private static bool IsAnyFileSelected(IReadOnlyList<PackageItem> children)
                     manifest.Version = Manifest.SupportedVersions.Max().ToString();
                 }
 
-                manifest.AddLibrary(new LibraryInstallationState
+                LibraryInstallationState libraryInstallationState = new LibraryInstallationState
                 {
                     LibraryId = PackageId,
                     ProviderId = selectedPackage.ProviderId,
                     DestinationPath = InstallationFolder.DestinationFolder,
                     Files = SelectedFiles.ToList()
-                });
+                };
 
-                await manifest.SaveAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                await Shell.ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
                 EnvDTE.Project project = VsHelpers.DTE.SelectedItems.Item(1)?.ProjectItem?.ContainingProject;
                 project?.AddFileToProjectAsync(_configFileName);
 
-                await _libraryCommandService.RestoreAsync(_configFileName, CancellationToken.None).ConfigureAwait(false);
+                RunningDocumentTable rdt = new RunningDocumentTable(Shell.ServiceProvider.GlobalProvider);
+                string configFilePath = Path.GetFullPath(_configFileName);
 
-                _dispatcher.Invoke(() =>
+                IVsTextBuffer document = rdt.FindDocument(configFilePath) as IVsTextBuffer;
```

Comment:
>document [](start = 30, length = 8)

can you rename this to better reflect the type?

Reply:
Sure, renamed it to textBuffer in version 10.

---
In reply to: [197931858](https://github.com/aspnet/LibraryManager/pull/246#discussion_r197931858) [](ancestors = 197931858)